src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -217,6 +217,21 @@ internal sealed override DeclarationScope EffectiveScope
         internal static SyntaxNode? GetDefaultValueSyntaxForIsNullableAnalysisEnabled(ParameterSyntax? parameterSyntax) =>
             parameterSyntax?.Default?.Value;
 
+        /// <summary>
+        /// Returns the bound default value syntax from the parameter, if it exists.
+        /// Note that this method will only return a non-null value if the
+        /// default value was supplied in syntax and not in metadata. Thus, there are cases
```

Comment:
>and not in metadata [](http://example.com/codeflow?start=49&length=19)

The reference to "metadata" probably is not appropriate for source only symbol. However, are there other scenarios that we should worry about? Should we be visiting all expressions in attributes as well? Constants can be referenced there. Also, can default parameter values be supplied via attributes (DefaultParameterValue, etc.)? It feels like this needs at least a follow up issue.

Reply:
I will update this comment. Yes, default parameter values can be supplied via attributes -- perhaps I can add that to the comment. There is currently a PR for tracking local constant usage in attributes, and I believe the work there would apply to the `DefaultParameterValue` attribute: https://github.com/dotnet/roslyn/pull/62158.